subsys/bluetooth/controller/ll_sw/ull_adv_iso.c

```diff
@@ -130,6 +135,22 @@ uint8_t ll_big_create(uint8_t big_handle, uint8_t adv_handle, uint8_t num_bis,
 	if (num_bis != 1) {
 		return BT_HCI_ERR_MEM_CAPACITY_EXCEEDED;
 	}
+
+	/* Add ACAD to AUX_SYNC_IND */
+	hdr_data.field_data = field_data;
+	field_data[0] = sizeof(struct pdu_big_info) + 2;
+	err = ull_adv_sync_pdu_set_clear(adv, ULL_ADV_PDU_HDR_FIELD_ACAD, 0U,
+					 &hdr_data, &ter_idx);
+	if (err) {
+		return err;
+	}
+
+	acad = (void *)sys_get_le32(&field_data[1]);
+	acad[0] = sizeof(struct pdu_big_info) + 1;
+	acad[1] = BT_DATA_BIG_INFO;
```

Comment:
This is where it is going in terms of implementation: https://github.com/zephyrproject-rtos/zephyr/pull/31009/commits/02e62ba4f365e29739006500f28c788436a0ade1#diff-46edd4480ca72c05f7bb026bdac384a2d48b03317361f2ea6ccdf70d15dc6769R197-R241

Passing by parameters or filling into a structure or passing the whole LLL context to fill inside `set_clear`, all have their disadvantage.

Reply:
I think having dedicated parameters or fields for passing the length and returning the address instead of reusing `field_data` would clear up a lot. Maybe the general pattern should be that `field_data` does not hold the data to be filled in on the way down, but the address where to write data on the way up.

So instead of writing (just as an example):
```
                cte_info.type = df_cfg->cte_type;
                cte_info.time = df_cfg->cte_length;
                pdu_data.field_data = (uint8_t *)&cte_info;
                ...
                err = ull_adv_sync_pdu_set_clear(adv, ULL_ADV_PDU_HDR_FIELD_CTE_INFO, 0, &pdu_data, &ter_idx);
```
the pattern could be
```
                err = ull_adv_sync_pdu_set_clear(adv, ULL_ADV_PDU_HDR_FIELD_CTE_INFO, 0, &pdu_data, &ter_idx);
                cte_info = pdu_data.field_data;                
                cte_info->type = df_cfg->cte_type;
                cte_info->time = df_cfg->cte_length;
```
This would then also fit the use case for the somewhat bigger big_info struct.

Reply:
The intention is to use the `ull_adv_aux_hdr_set_clear` and `ull_adv_sync_pdu_set_clear` (in the future combine them into one function) to take-in variable number of data for numerous header fields (bitmap for set/clear, say adding adv and init addresses, may be other parameters together). Hence the field_data is length encoded (like the AD data format, length-data). This is done to save on execution time updating the PDU.

Let me review so far, if array of lengths can be passed and then return array of pointers that can be used to fill in. (But this will not happen until I get to the point I have an initial samples of ISO broadcaster and ISO receiver)

The current implementation of passing length+value for simple values (AdvA, TargetA, ADI, AuxPtr, TxPower and Host AD Data) and returning pointers for struct values (CTEInfo, SyncInfo, and ACAD).

This means, we keep as-is for AdvA, TargetA, ADI, AuxPtr, TxPower and Host AD Data; and CTEInfo and SyncInfo be ported to populating the return address, do you agree?

Reply:
I would prefer to use the same pattern for everything and I think it would eventually lead to simpler, more efficient code. But I can live with distinguishing between "simple" and "complex" cases.

Reply:
@wopu-ot as communicated to you few weeks back in our status meeting, I tend to agree with passing down field bitmap and passing up address as stated in your comment here https://github.com/zephyrproject-rtos/zephyr/pull/32622#discussion_r587425119

@andrzej-kaczmarek FYI.

Reply:
@wopu-ot Updated with the sync_info population by caller. Please review.